substrate/primitives/src/block.rs

```diff
@@ -84,40 +88,49 @@ impl Slicable for Digest {
 	}
 }
 
-/// The body of a block is just a bunch of transactions.
-pub type Body = Vec<Transaction>;
+impl NonTrivialSlicable for Digest { }
 
-/// A Substrate relay chain block.
-#[derive(PartialEq, Eq, Clone)]
-#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
-pub struct Block {
-	/// The block header.
-	pub header: Header,
-	/// All relay-chain transactions.
-	pub transactions: Body,
-}
+/// Generic types to be specialised later.
+pub mod generic {
+	use super::{Header, Slicable, Input, NonTrivialSlicable, Vec};
 
-impl Slicable for Block {
-	fn decode<I: Input>(input: &mut I) -> Option<Self> {
-		Some(Block {
-			header: try_opt!(Slicable::decode(input)),
-			transactions: try_opt!(Slicable::decode(input)),
-		})
+	/// A Block - this is generic for later specialisation in particular runtimes.
+	#[derive(PartialEq, Eq, Clone)]
+	#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Debug))]
+	pub struct Block<Transaction: PartialEq + Eq + Clone> {
```

Comment:
See the changes in #77 : rather than a simple `Vec<Transaction>` for some transaction type, the body is certain fields like timestamp, parachains (to be done), and _then_ transactions. It serializes as a `Vec<Vec<u8>>` (soon to be `Vec<Extrinsic>`). 

The generalization change here isn't useful for that kind of model.